contrib/epee/include/net/abstract_tcp_server2.inl

```diff
@@ -269,8 +269,6 @@ PRAGMA_WARNING_DISABLE_VS(4355)
     //_dbg3("[sock " << socket().native_handle() << "] add_ref, m_peer_number=" << mI->m_peer_number);
     CRITICAL_REGION_LOCAL(self->m_self_refs_lock);
     //_dbg3("[sock " << socket().native_handle() << "] add_ref 2, m_peer_number=" << mI->m_peer_number);
-    if(m_was_shutdown)
-      return false;
```

Comment:
AFAICT, this would mean operations get started on shutdown connections. Why is this needed, rather then let those operations fail in the first place ?

Reply:
Operations could be requested after shutdown before or after this change. The lock was immediately released a few lines below, with requests for sending data occurring after that. This might cause some more calls into the send function to fail, but it doesn't appear incorrect.

There's still a "race" to the `add_invoke_response_handler` call but that should only delay cleanup of the object until the async invoke timeout is triggered.